resizable-circle/demo.js

```diff
@@ -0,0 +1,134 @@
+ */
+function createResizableCircle(map) {
+  var circle = new H.map.Circle(
+        {lat: 50, lng: 22.8},
+        85000,
+        {
+          style: {fillColor: 'rgba(250, 250, 0, 0.7)', lineWidth: 0}
+        }
+      ),
+      circleOutline = new H.map.Polyline(
+        circle.getGeometry().getExterior(),
+        {
+          style: {lineWidth: 8, strokeColor: 'rgba(255, 0, 0, 0)'}
+        }
+      ),
+      circleGroup = new H.map.Group({
+        volatility: true, // mark the group as volatile for smooth dragging of all it's objects
+        objects: [circle, circleOutline]
+      }),
+      circleTimeout;
+
+  // ensure that the objects can receive drag events
+  circle.draggable = true;
+  circleOutline.draggable = true;
+
+  // extract first point of the circle outline polyline's LineString and
+  // push it to the end, so the outline has a closed geometry
+  circleOutline.getGeometry().pushPoint(circleOutline.getGeometry().extractPoint(0));
+
+  // add group with circle and it's outline (polyline)
+  map.addObject(circleGroup);
+
+  // event listener for circle group to show outline (polyline) if moved in with mouse (or touched on touch devices)
+  circleGroup.addEventListener('pointerenter', function(evt) {
+    var group = this,
+        circleOutline = group.getObjects()[1],
+        currentStyle = circleOutline.getStyle(),
+        newStyle = currentStyle.getCopy({
+          strokeColor: 'rgb(255, 0, 0)'
+        });
+
+    if (circleTimeout) {
+      clearTimeout(circleTimeout);
+      circleTimeout = null;
+    }
+    // show outline
+    circleOutline.setStyle(newStyle);
+  }, true);
+
+  // event listener for circle group to hide outline if moved out with mouse (or released finger on touch devices)
+  // the outline is hidden on touch devices after specific timeout
+  circleGroup.addEventListener('pointerleave', function(evt) {
+    var group = this,
+        circleOutline = group.getObjects()[1],
+        currentStyle = circleOutline.getStyle(),
+        newStyle = currentStyle.getCopy({
+          strokeColor: 'rgba(255, 0, 0, 0)'
+        }),
+        timeout = (evt.currentPointer.type == 'touch') ? 1000 : 0;
+
+    circleTimeout = setTimeout(function() {
+      circleOutline.setStyle(newStyle);
+    }, timeout);
+    document.body.style.cursor = 'default';
+  }, true);
+
+  // event listener for circle group to change the cursor if mouse position is over the outline polyline (resizing is allowed)
+  circleGroup.addEventListener('pointermove', function(evt) {
+    if (evt.target instanceof H.map.Polyline) {
+      document.body.style.cursor = 'pointer';
+    } else {
+      document.body.style.cursor = 'default'
+    }
+  }, true);
+
+  // event listener for circle group to resize the geo circle object if dragging over outline polyline
+  circleGroup.addEventListener('drag', function(evt) {
+    var pointer = evt.currentPointer,
+        group = this,
+        circle = group.getObjects()[0],
```

Comment:
I'd suggest that for the simplicity's sake we use the variables from the outer scope, then we don't neet group=this; asisignment

Reply:
done